### PR TITLE
chore: add webhook validation for `shared_buffers`

### LINKED ISSUE #2664: CNPG should validate memory resources settings against the `shared_memory` Postgres parameter
In #2661, we added a validator to catch some possible mistakes in resource definition.

We can expand it to catch more obvious mistakes in the memory settings, comparing them to the `shared_memory` Postgres configuration parameter. 

If defined, the memory resources must be greater than the Postgres shared memory.

We could also talk about it if we want to let the user know (with a warning) if the values are too narrow, but that would be a heuristic and may require another ticket to introduce it.

### PR DESCRIPTION
**Pull request by armru**
Partially Closes #2664

### CODE REVIEW

File: api/v1/cluster_webhook.go [hunk 1]
```diff
@@ -1009,6 +1028,17 @@ func (r *Cluster) validateConfiguration() field.ErrorList {
 		}
 	}
 
+	if value := r.Spec.PostgresConfiguration.Parameters[sharedBuffersParameter]; value != "" {
+		if _, err := parsePostgresQuantityValue(value); err != nil {
```
Comment (YanniHu1996) [inline api/v1/cluster_webhook.go:1032]: the error is hidden, it is better to let the user know what went wrong,  or what is the correct format

**Review by YanniHu1996 — APPROVED** [review]

File: api/v1/cluster_webhook.go (hunk 1 above)
Comment (armru) [reply api/v1/cluster_webhook.go:1032]: Updated the message, is it any better?

**Review by YanniHu1996 — APPROVED** [review]

**Review by litaocdl — APPROVED** [review]

**Comment by litaocdl** [comment]
run from fork https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/6297393445

File: api/v1/cluster_webhook.go [hunk 2]
```diff
@@ -944,25 +948,40 @@ func (r *Cluster) validateImagePullPolicy() field.ErrorList {
...
-		memoryRequestGtThanLimit := r.Spec.Resources.Requests.Memory().Cmp(*r.Spec.Resources.Limits.Memory()) > 0
+	memoryRequest := r.Spec.Resources.Requests.Memory()
+	rawSharedBuffer := r.Spec.PostgresConfiguration.Parameters[sharedBuffersParameter]
+	if !memoryRequest.IsZero() && rawSharedBuffer != "" {
+		if sharedBuffers, err := parsePostgresQuantityValue(rawSharedBuffer); err == nil {
+			if memoryRequest.Cmp(sharedBuffers) < 0 {
+				result = append(result, field.Invalid(
+					field.NewPath("spec", "resources", "requests", "memory"),
+					memoryRequest.String(),
+					"Memory request is lower than the shared_buffer value",
```
Comment (gbartolini) [inline api/v1/cluster_webhook.go:972]:
```suggestion
					"Memory request is lower than PostgreSQL `shared_buffers` value",
```

File: api/v1/cluster_webhook_test.go [hunk 3]
```diff
@@ -1052,6 +1052,20 @@ var _ = Describe("configuration change validation", func() {
 		}
 		Expect(clusterNew.validateConfiguration()).To(HaveLen(1))
 	})
+
+	It("should detect an invalid shared_buffer value", func() {
```
Comment (gbartolini) [inline api/v1/cluster_webhook_test.go:1056]:
```suggestion
	It("should detect an invalid `shared_buffers` value", func() {
```

**Review by gbartolini — APPROVED** [review]

**Pull request merged**